crates/meilisearch-auth/src/store.rs

```diff
@@ -328,30 +297,24 @@ where
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
-struct KeyActions {
-    actions: u64, // Bitflags for actions
+struct KeyMasks {
+    bit_mask: u32, // Bitflags for bit_maks
     indexes: Vec<IndexUidPattern>,
     expires_at: Option<OffsetDateTime>,
 }
 
-impl KeyActions {
-    fn new(
-        actions: &HashSet<Action>,
-        indexes: &[IndexUidPattern],
-        expires_at: Option<OffsetDateTime>,
-    ) -> Self {
-        let mut bitflags = 0u64;
-        for (index, action_flag) in enum_iterator::all::<Action>().enumerate() {
-            if actions.contains(&action_flag) {
-                bitflags |= 1 << index;
-            }
+impl KeyMasks {
+    fn new(bit_mask: u32, indexes: &[IndexUidPattern], expires_at: Option<OffsetDateTime>) -> Self {
+        let mut bitflags = 0u32;
+        for action in bit_mask {
+            bitflags |= action.bits();
         }
-        Self { actions: bitflags, indexes: indexes.to_vec(), expires_at }
+        Self { bit_mask: bitflags, indexes: indexes.to_vec(), expires_at }
```

Comment:
Because in the previous comment we deconstruct the `Key` structure, we take the ownership of the inner values, so the `.to_vec` should not be necessary anymore, which would save memory allocation.
```suggestion
        Self { bit_mask: bitflags, indexes, expires_at }
```

Moreover, if you keep the same name variable for `bitflags`/`bit_mask` it would ease the readability and Rust would thank you by allowing you to precise the name only once:

```suggestion
        Self { bitflags, indexes, expires_at }
```
(`KeyMasks::bit_mask` renamed into `KeyMasks::bitflags`)

Reply:
thanks for this guidence I have implemented this